File: publish/project/androidproj/pushmsg/src/main/java/com/fy/game/pushmsg/IntervalChecker.java
package com.fy.game.pushmsg;

/**
 * Created by bill on 2017/7/11.
 */

public class IntervalChecker {
    private long lastTime = -0x7fffffff;
    private long interval = 0;

    public IntervalChecker(long intervalSecond) {
        interval = intervalSecond * 1000;
    }

    public boolean isComing() {
        if (System.currentTimeMillis() - lastTime < interval) {
            return false;
        } else {
            lastTime = System.currentTimeMillis();
            return true;
        }
    }
}

File: publish/project/Assets/Scripts/utils/IntervalChecker.cs
public class IntervalChecker
{
    private float lastTime = 0;
    private float interval = 0;
    public IntervalChecker(float intervalSecond)
    {
        interval = intervalSecond;
    }
    public bool check()
    {
        if (UnityEngine.Time.realtimeSinceStartup - lastTime < interval)
        {
            return false;
        }
        else
        {
            lastTime = UnityEngine.Time.realtimeSinceStartup;
            return true;
        }
    }
}